components/tools/OmeroJava/test/integration/chown/PermissionsTest.java

```diff
@@ -270,44 +350,140 @@ public void testChownAnnotationPrivate(boolean isDataOwner, boolean isAdmin, boo
             return;
         }
 
-        /* check that the objects' ownership is all as expected */
+        /* check that the objects' ownership is all as expected, i.e. the non-doubly linked
+         * annotations belong to recipient, the doubly-linked annotations belong to importer */
 
         final Set<Long> imageLinkIds = new HashSet<Long>();
 
         logRootIntoGroup(dataGroupId);
         assertOwnedBy(image, recipient);
         for (final IObject annotation : imageAnnotations) {
-            if (annotation instanceof TagAnnotation) {
+            if (annotation instanceof TagAnnotation || annotation instanceof FileAnnotation || 
+                annotation instanceof MapAnnotation) {
                 assertOwnedBy(annotation, importer);
             } else if (annotation instanceof ImageAnnotationLink) {
                 imageLinkIds.add(annotation.getId().getValue());
             } else {
+
                 assertOwnedBy(annotation, recipient);
             }
         }
+
+        for (final IObject annotation : imageAnnotations2) {
+        	assertOwnedBy(annotation, recipient);
+        }
+
         assertOwnedBy(tagLinksOnOtherImage, importer);
+        assertOwnedBy(fileAnnLinksOnOtherImage, importer);
+        assertOwnedBy(mapAnnLinksOnOtherImage, importer);
 
-        /* check that the image's links to the tags that were also linked to the other image were deleted */
+        /* check that the image's links to the tags and FileAnnotations that were also linked to the other image were deleted */
 
         final String query = "SELECT COUNT(id) FROM ImageAnnotationLink WHERE id IN (:ids)";
         final ParametersI params = new ParametersI().addIds(imageLinkIds);
         final List<List<RType>> results = iQuery.projection(query, params);
         final long remainingLinkCount = ((RLong) results.get(0).get(0)).getValue();
-        Assert.assertEquals(remainingLinkCount, imageLinkIds.size() - tagLinksOnOtherImage.size());
+        final long deletedLinkCount = tagLinksOnOtherImage.size() + fileAnnLinksOnOtherImage.size() + mapAnnLinksOnOtherImage.size();
+        Assert.assertEquals(remainingLinkCount, imageLinkIds.size() - deletedLinkCount);
+        
+        /* chown the first tag set */
+        init(chowner);
+        final Chown2 chown2 = Requests.chown().target(tagsets.get(0)).toUser(recipient.userId).build();
+        
+        doChange(client, factory, chown, isExpectSuccess);
+
+        if (!isExpectSuccess) {
+            return;
```

Comment:
Would it be better to `Assert.fail()` in that case? Just `return` implies the test passed, doesn't it?